src/merge.py

```diff
@@ -16,7 +16,10 @@ def merge(x: list[int], y: list[int]) -> list[int]:
     # FIXME: fill out the loop so you merge the lists
     # until one of them is empty
     while i < len(x) and j < len(y):
-        break  # FIXME: you shouldn't just break here
-    # At least one of the lists is empty now. Copy the
-    # remainder of the other into z.
-    return z
+        if x[i] < y[j]:
+            z.append(x[i])
+            i += 1
+        else:
+            z.append(y[j])
+            j += 1
+    return z + x[i:len(x)] + y[j:len(y)]
```

Comment:
Two things here.

1) When you use slicing, as in `x[i:j]`, you don't need to write 0 for `i` if it is zero and you don't need to write `len(x)` for `j` if it is `len(x)`. You can leave out the numbers, and those will be used as defaults. That is, `x[:j]` is the same as `x[0:j]` and `x[i:]` is the same as `x[i:len(x)]`. There is nothing wrong with what you do, using the defaults is just shorter and often easier to read.

2) When you write `z + x[i:] + y[j:]`, you are copying all three lists into a new list. There is nothing wrong with that, but if `z` is long it is wasting time. (You have to copy the other two anyway, so there is nothing to save there, but there might be a lot to save for `z`). Instead you can do

```python
z.extend(x[i:])
z.extend(y[j:])
return z
```

Reply:
Understood! Thank you very much